lib/spack/spack/cmd/external.py

```diff
@@ -148,10 +152,21 @@ def _spec_is_valid(spec):
 
 
 def external_find(args):
+    # Construct the list of possible packages to be detected
+    packages_to_check = []
+
+    # If tags have been passed, add the corresponding packages
+    if args.tags:
+        for pkg in spack.repo.path.packages_with_tags(*args.tags):
+            packages_to_check.append(spack.repo.get(pkg))
```

Comment:
I'm not sure this isn't the behavior we want.

What should `spack external find -t dev-tools autoconf openmpi` do? Should it

1. find `autoconf`, because that is the listed package that satisfies the tag
2. find `openmpi` + all packages tagged as dev-tools.

I would lean towards the former.

Reply:
I was leaning towards the latter obviously, but have no strong preference. If tags are considered filters, like you propose, the most common use case would still remain:
```console
$ spack external find -t build-tools 
```
I'll do the change and ping when done